### PR TITLE
iff->if or if and only if 

### PR DESCRIPTION
**Pull request by saroh**
rationale : has_XXX, is_XXX functions are nearly always defined as a iff in code. Other bolean returns you may need to specify when to return what value.

#1236

### CODE REVIEW

**Comment by fulmicoton** [comment]
iff means if and only if. It has confused people in the past so I opened an issue to replace it with if and only if
https://github.com/quickwit-oss/tantivy/issues/1236

I guess if alone is ok too where it is not amibguous.
https://en.wikipedia.org/wiki/If_and_only_if#:~:text=In%20logic%20and%20related%20fields,true%20or%20both%20are%20false.

**Comment by saroh** [comment]
Oh I had the feelingh it may have ben something else, I'm re-writing the PR :)

**Comment by saroh** [comment]
So I feel I could put put `if` on functions/methods that implement an obvious single responsability, so most of your code. There are a few interfaces where the `if and only if` should be stressed imo, as a guideline for the implementer: methods which answer a boolean but are not obvious booleans (eg. advance vs is_something). Or exotic truth values such as bloomfilters https://en.wikipedia.org/wiki/Bloom_filter

File: src/collector/top_collector.rs [hunk 1]
```diff
@@ -173,8 +173,7 @@ impl<T: PartialOrd + Clone> TopSegmentCollector<T> {
             .collect()
     }
 
-    /// Return true iff at least K documents have gone through
-    /// the collector.
+    /// Return true if more documents have been collected than the limit.
```
Comment (saroh) [inline src/collector/top_collector.rs:176]: wording might be iffy here :)

File: src/termdict/fst_termdict/termdict.rs [hunk 2]
```diff
@@ -143,7 +143,8 @@ impl TermDictionary {
         Ok(self.fst_index.get(key))
     }
 
-    /// Returns the term associated to a given term ordinal.
+    /// Stores the term associated to a given term ordinal in
+    /// a `bytes` buffer.
     ///
```
Comment (saroh) [inline src/termdict/fst_termdict/termdict.rs:148]: tocheck, out of PR scope ;)

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
Thank you @saroh !

**Pull request merged**